starknet_py/net/client_models.py

```diff
@@ -279,13 +312,31 @@ class BlockStatus(Enum):
 @dataclass
 class PendingStarknetBlock:
     """
-    Dataclass representing pending block on Starknet.
+    Dataclass representing a pending block on Starknet.
     """
 
     transactions: List[Transaction]
-    timestamp: Optional[int] = None
-    sequencer_address: Optional[int] = None
-    parent_hash: Optional[int] = None
+    parent_block_hash: int
+    timestamp: int
+    sequencer_address: int
+    # TODO (#1179): this field should be required
+    l1_gas_price: Optional[ResourcePrice] = None
+    # TODO (#1179): this field should be required
+    starknet_version: Optional[str] = None
+
+
+@dataclass
+class PendingStarknetBlockWithTxHashes:
+    """
+    Dataclass representing a pending block on Starknet containing transaction hashes.
+    """
+
+    transactions: List[int]
+    parent_block_hash: int
+    timestamp: int
+    sequencer_address: int
+    l1_gas_price: Optional[ResourcePrice] = None
+    starknet_version: Optional[str] = None
```

Comment:
nit: I guess to be consistent a comment `TODO (#1179): this field should be required` should be also added here